Ginger/GingerCore/ALM/JIRA/Bll/JiraExportManager.cs

```diff
@@ -281,6 +321,114 @@ public bool ExecuteDataToJira(BusinessFlow bizFlow, PublishToALMConfig publishTo
+                {
+                    resultFlag = false;
+                }
+            }
+            if (resultFlag)
+                result = "Export has been finished successfully";
+            else
+                result = "Error Has been Happened while export to execution results to Jira-Zephyr";
+            return resultFlag;
+        }
+
+        public void AssignDefectsToZephyrExecutions(Dictionary<Guid, Dictionary<string, string>> defectsForOpening, Dictionary<Guid, string> defectsOpeningResults)
+        {
+            try
+            {
+                if ((defectsForOpening != null) && (defectsForOpening.Count > 0))
+                {
+                    List<JiraZephyrExecution> executionList = ((JiraManagerZephyr)jiraRepObj.TestAlmManager()).GetZephyrExecutionList(ALMCore.DefaultAlmConfig.ALMUserName, ALMCore.DefaultAlmConfig.ALMPassword,
+                                                                                                                                            ALMCore.DefaultAlmConfig.ALMServerURL, string.Empty, string.Empty,
+                                                                                                                                            ALMCore.DefaultAlmConfig.ALMProjectKey, defectsForOpening.First().Value["BFExternalID1"], string.Empty, string.Empty,
+                                                                                                                                            string.Empty, string.Empty, defectsForOpening.First().Value["BFExternalID2"] == null ? string.Empty : defectsForOpening.First().Value["BFExternalID2"]).DataResult;
+
+                    foreach (KeyValuePair<Guid, string> defectOpeningResult in defectsOpeningResults)
+                    {
+                        KeyValuePair<Guid, Dictionary<string, string>> currentDefect = defectsForOpening.Where(z => z.Key == defectOpeningResult.Key).FirstOrDefault();
+                        JiraZephyrExecution currentActivitiesGroupExecution = executionList.Where(z => z.IssueId.ToString() == currentDefect.Value["ActivityGroupExternalID"]).FirstOrDefault();
+                        if (currentDefect.Key != null)
+                        {
+                            JiraZephyrResponse jiraZephyrResponse = ((JiraManagerZephyr)jiraRepObj.TestAlmManager()).UpdateExecutionWithDefects(ALMCore.DefaultAlmConfig.ALMUserName, ALMCore.DefaultAlmConfig.ALMPassword, ALMCore.DefaultAlmConfig.ALMServerURL,
+                                                                                                                                                new List<long>() { currentActivitiesGroupExecution.Id },
+                                                                                                                                                new List<string>() { defectOpeningResult.Value }).DataResult;
+                        }
+                    }
+                }
+            }
+            catch
```

Comment:
Codacy found an issue: [Handle the exception or explain in a comment why it can be ignored.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6280018)